pkg/storage/module/upload.go

```diff
@@ -14,16 +15,30 @@ import (
 const numFiles = 3
 
 // Uploader takes a stream and saves it to the blob store under a given path
-type Uploader func(ctx context.Context, path, contentType string, stream io.Reader) error
+type Uploader func(ctx context.Context, path, contentType string, stream Stream) error
```

Comment:
Can all Storage save by `path`? Shouldn't the uploader take a `module, version` and then the storage itself determines how to combine them into a path, or maybe not combine them at all? This would be very necessary when we want to fix `module.Checker` to be able to tell us how many files a particular version has, or how many versions a particular module has.

Reply:
Btw the comment above is out of this PR's scope, but just wanted to make peeps aware of potential changes to it so that we don't break everyone twice